### DISCUSSION TITLE
How to hide adapter selection for the user when using just 1 provider

### DISCUSSION THREAD

**Discussion by fediazgon** [nextauthjs/next-auth#7261] [Help]
### Question 💬

Let's say I'm only planning to use the Auth0 provider in my app. What's the blessed way of hiding the selection for the user when /api/auth/signin is called?

I was thinking that one possible solution would be to redirect the user to a custom login page that will call the signIn("auth0") method on render. Is there a simpler way that I'm missing?

### How to reproduce ☕️

N/A

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by paul-vd** [comment]
Same questions here: https://github.com/nextauthjs/next-auth/discussions/4790

Currently I'm handeling this in a `useEffect` with a custom login page.

```jsx
"use client";

import { signIn } from "next-auth/react";
import { useEffect } from "react";

export default function LoginPage() {
  useEffect(() => {
    signIn("auth0");
  }, []);

  return null;
}

```

This is of course not ideal as it would have to load this extra useless page, and you will also have to think about handeling callbacks which should not redirect,

After inspecting the source code, there is not really any way of getting the provider URL unless you reconstruct a custom request which is compatible with the next-auth [internalRequest](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/core/index.ts#L54-L86)

Some related resources of how the `signIn` function works behind the scenes:
1. It sends a fetch to the `api/auth/signin/auth0` endpoint (or which ever provider you pass to the `signIn`)
1. The `AuthHandler` detects which request it should handle, in this case [`method=post`](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/core/index.ts#L239) 
1. It then checks which action to perform, in this case [`signin`](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/core/index.ts#L241)
1. It then runs the [signin route](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/core/index.ts#L244-L248) to return the redirect url which it is fetched with the oauth [getAuthorizationUrl](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/core/lib/oauth/authorization-url.ts#L16-L63).

Other than that, without recreating a fake request, currently I don't think there is any entry point to achieve this. It would be a good feature though for instances where there is only 1 auth provider.

**Reply by paul-vd** [reply]
You could also build the auth url manually by doing something like this, and using the auth0 provider config

```js
function redirectUrl(state: string) {
  // first we create a new URL instance using our auth0.domain as initial value
  let url = new URL(auth0.domain);
  // then we set the pathname to `/authorize`
  url.pathname = "/authorize";
  // and now we start adding search params (aka the query string), we need a response_type as code
  url.searchParams.set("response_type", "code");
  // we need the client_id
  url.searchParams.set("client_id", auth0.clientId);
  // the callback URL as redirect_uri
  url.searchParams.set("redirect_uri", auth0.callbackUrl);
  // the scope of the data we want, this is a list of words with a blank space between them
  url.searchParams.set("scope", "openid profile email");
  // the last one is the state, this is our UUID, it can technically be any random string
  url.searchParams.set("state", state);
  // finally we return the URL as a string
  return url.toString();
}
```
